.github/workflows/build.yml

```diff
@@ -162,13 +162,23 @@ jobs:
           retry_wait_seconds: 10
           command: docker pull docker.pkg.github.com/apache/incubator-nuttx-testing/nuttx-ci-linux
 
+      - name: Export NuttX Repo SHA
+        run:  echo "::set-env name=nuttx_sha::`git -C sources/nuttx rev-parse HEAD`"
+      - name: Refresh Git Credentials
+        uses: actions/checkout@v2
+        with:
+          repository: apache/incubator-nuttx
+          ref: ${{ env.nuttx_sha }}
+          path: sources/nuttx
+          fetch-depth: 1
+      - name: Get Tags for NuttX Repo
+        run: git -C sources/nuttx fetch --tags
```

Comment:
I may ask the same question before: does it make sense to move the tag fetch to 'Fetch-Source->Checkout nuttx repo'?

Reply:
Causes a huge slow down because the cache becomes substantially larger.  I want to implement a better solution later this week that goes back to just doing the checkouts in the build job but computes the refs in the shared job, but I want to unblock the CI for now.

Reply:
I am curious why tag can increase the size so much? as I understand that tag is just a simple marker.

Reply:
because we loose the clone of depth=1 when you pull down tags since it has to pull the full history :/

Reply:
Just saw this from https://github.com/actions/checkout:
## Fetch all tags

```yaml
- uses: actions/checkout@v2
- run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
```

## Fetch all branches

```yaml
- uses: actions/checkout@v2
- run: |
    git fetch --no-tags --prune --depth=1 origin +refs/heads/*:refs/remotes/origin/*
```

## Fetch all history for all tags and branches

```yaml
- uses: actions/checkout@v2
- run: |
    git fetch --prune --unshallow
```
Can we use them to fix the problem without dropping the speed.